code/lib/cli/src/upgrade.ts

```diff
@@ -142,8 +142,10 @@ export const doUpgrade = async ({
   if (!isCanary && lt(currentVersion, beforeVersion)) {
     throw new UpgradeStorybookToLowerVersionError({ beforeVersion, currentVersion });
   }
+
   if (!isCanary && eq(currentVersion, beforeVersion)) {
-    throw new UpgradeStorybookToSameVersionError({ beforeVersion });
+    // Not throwing, as the beforeVersion calculation doesn't always work in monorepos.
+    logger.error(new UpgradeStorybookToSameVersionError({ beforeVersion }));
```

Comment:
Should we log it as a warning instead if it isn't failing?

Reply:
Yeah, might be better indeed.